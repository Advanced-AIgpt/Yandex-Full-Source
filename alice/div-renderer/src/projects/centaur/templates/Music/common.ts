import {
    ContainerBlock,
    Div,
    FixedSize,
    IDivAction,
    ImageBackground,
    ImageBlock, MatchParentSize,
    SolidBackground,
    TextBlock,
} from 'divcard2';
import { colorBlackOpacity70, offsetFromEdgeOfScreen } from '../../style/constants';
import { compact } from 'lodash';
import EmptyDiv from '../../components/EmptyDiv';
import { text28m } from '../../style/Text/Text';
import { directivesAction } from '../../../../common/actions';
import { centaurMusicPlay } from '../../../../common/actions/server/musicActions';
import MusicPlaying from './MusicPlaying/MusicPlaying';
import { IRequestState } from '../../../../common/types/common';
import LikeButton from './components/LikeButton';

interface IGetPlaylistItemProps {
    title: string;
    subtitle?: string;
    objectId: string;
    objectType: string;
    startFromTrackId?: string;
    isCurrent?: boolean;
    imageUrl: string;
    imageBackground?: string;
    requestState: IRequestState;
    isLiked?: boolean;
}

function CoverWithBackground(imageUrl: string, background: string, action: IDivAction, isCurrent: boolean, requestState: IRequestState) {
    return new ContainerBlock({
        width: new FixedSize({ value: 80 }),
        height: new FixedSize({ value: 80 }),
        alignment_vertical: 'top',
        actions: [action],
        margins: {
            right: 24,
        },
        border: {
            corner_radius: 40,
        },
        content_alignment_horizontal: 'center',
        content_alignment_vertical: 'center',
        background: compact([
            new SolidBackground({
                color: background,
            }),
        ]),
        orientation: 'overlap',
        items: compact([
            new ImageBlock({
                image_url: imageUrl,
                width: new FixedSize({ value: 48 }),
                height: new FixedSize({ value: 48 }),
            }),
            isCurrent && new EmptyDiv({
                width: new FixedSize({ value: 80 }),
                height: new FixedSize({ value: 80 }),
                background: [
                    new SolidBackground({
                        color: colorBlackOpacity70,
                    }),
                ],
            }),
            isCurrent && MusicPlaying(requestState),
        ]),
    });
}

function CoverWithoutBackground(imageUrl: string, action: IDivAction, isCurrent: boolean, requestState: IRequestState) {
    return new ContainerBlock({
        width: new FixedSize({ value: 80 }),
        height: new FixedSize({ value: 80 }),
        alignment_vertical: 'top',
        content_alignment_horizontal: 'center',
        content_alignment_vertical: 'center',
        actions: [action],
        margins: {
            right: 24,
        },
        border: {
            corner_radius: 20,
        },
        background: compact([
            new ImageBackground({
                image_url: imageUrl,
            }),
            isCurrent && new SolidBackground({
                color: colorBlackOpacity70,
            }),
        ]),
        items: [
            isCurrent ? MusicPlaying(requestState) : new EmptyDiv(),
        ],
    });
}

export function getPlaylistItem({
    isCurrent = false,
    objectType,
    subtitle,
    objectId,
    title,
    startFromTrackId,
    imageUrl,
    imageBackground,
    requestState,
    isLiked,
}: IGetPlaylistItemProps): {
    options: Partial<ConstructorParameters<typeof ContainerBlock>[0]>;
    items: Div[];
} {
    const openAction = {
        log_id: 'play_music_click',
        url: directivesAction(centaurMusicPlay({
            disableNlg: true,
            objectType,
            objectId,
            startFromTrackId,
        })),
    };

    return {
        options: {
            orientation: 'horizontal',
            paddings: {
                top: 28,
                left: offsetFromEdgeOfScreen,
                bottom: 28,
                right: offsetFromEdgeOfScreen,
            },
            content_alignment_vertical: 'center',
        },
        items: compact([
            imageBackground ?
                CoverWithBackground(imageUrl, imageBackground, openAction, isCurrent, requestState) :
                CoverWithoutBackground(imageUrl, openAction, isCurrent, requestState),
            new ContainerBlock({
                width: new MatchParentSize({ weight: 1 }),
                actions: [openAction],
                items: compact([
                    new TextBlock({
                        ...text28m,
                        text: title,
                    }),
                    subtitle && new TextBlock({
                        ...text28m,
                        text: subtitle,
                        alpha: 0.5,
                    }),
                ]),
            }),
            typeof isLiked !== 'undefined' && LikeButton({ isLiked, trackId: startFromTrackId || '' }),
        ]),
    };
}
