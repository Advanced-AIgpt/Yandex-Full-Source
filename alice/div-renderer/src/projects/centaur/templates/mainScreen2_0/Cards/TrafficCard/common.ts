import { ITrafficForecast } from './types';

export function sortTimeOfDay(originalArray: Readonly<ITrafficForecast[]>): ITrafficForecast[] {
    const forecastsFirst = [];
    const forecastsLast = [];

    for (const forecastElement of originalArray) {
        if (forecastElement.time === null || typeof forecastElement.time === 'undefined') {
            continue;
        }

        if (forecastsFirst.length === 0) {
            forecastsFirst.push(forecastElement);
        } else if (forecastsLast.length > 0) {
            forecastsLast.push(forecastElement);
        } else if (forecastElement.time - (forecastsFirst[forecastsFirst.length - 1].time as number) === 1) {
            forecastsFirst.push(forecastElement);
        } else {
            forecastsLast.push(forecastElement);
        }
    }

    return [...forecastsLast, ...forecastsFirst];
}
