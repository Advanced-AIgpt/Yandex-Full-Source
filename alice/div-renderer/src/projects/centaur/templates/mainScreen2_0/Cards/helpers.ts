export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
