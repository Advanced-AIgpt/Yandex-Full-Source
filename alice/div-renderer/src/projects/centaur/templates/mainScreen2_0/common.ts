import { IDivAction } from 'divcard2';
import { EnumLayer } from '../../actions/client';
import { CrutchLoader } from '../../components/Loader/Loader';

interface IPutLoaderToActionOptions {
    action?: string | null,
    layer: EnumLayer,
    actionLogId?: string,
}

export const putLoaderToAction = ({
    action,
    layer,
    actionLogId = 'action_with_loader',
}: IPutLoaderToActionOptions): IDivAction[] | undefined => {
    if (!action) {
        return undefined;
    }
    return [
        CrutchLoader(layer),
        {
            log_id: actionLogId,
            url: action,
        },
    ];
};
