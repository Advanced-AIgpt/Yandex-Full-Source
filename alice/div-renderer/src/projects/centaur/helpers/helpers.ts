/* eslint-disable camelcase */
import { sortedIndexBy } from 'lodash';
import { format, isSameDay } from 'date-fns';
import { ru } from 'date-fns/locale';
import { Div, IDivData, TextBlock } from 'divcard2';
import { IRequestState } from '../../../common/types/common';
import { getTemplates, templatesRegistry } from '../../../registries/common';
import { centaurTemplatesClass } from '../index';
import { Palette } from './palette';

export { default as pluralize } from '@yandex-int/i18n/lib/plural/ru';
type TextProps = Pick<ConstructorParameters<typeof TextBlock>[0], 'text_color'|'font_size'|'font_weight'|'line_height'>;

export const textBreakpoint =
    (...breakpoints: [number, TextProps][]) =>
        (text: string) => {
            const idx = Math.min(sortedIndexBy(breakpoints, [text.length, {}], 0), breakpoints.length - 1);
            return { text, ...breakpoints[idx][1] };
        };

export const formatDate = (d: Date | string | number, f: string) => format(new Date(d), f, { locale: ru });

export const textCommand = (c: string) => `dialog://text_command?query=${encodeURI(c)}`;

export const updatedAt = (d: Date) => {
    const now = new Date();
    return `Обновлен ${isSameDay(now, d) ? 'сегодня' : formatDate(d, 'dd.MM')}`;
};

export const insertBetween = <I extends unknown, S extends unknown>(itemList: I[], separator: S) => {
    return itemList.reduce<(I | S)[]>((result, item, index) => {
        if (index !== 0) {
            result.push(separator);
        }

        result.push(item);

        return result;
    }, []);
};

export interface ITemplateCard {
    templates: {
        [name: string]: Div;
    };
    palette?: Palette;
    card: IDivData;
}

// Дива верхнего уровня. Замена для new `TemplateCard()`
export const TopLevelCard = (card: Omit<IDivData, 'variable_triggers' | 'variables'>, requestState: IRequestState, palette?: Palette) => {
    const templates = Object.assign(
        getTemplates(requestState.localTemplates, templatesRegistry),
        centaurTemplatesClass.getUsedTemplates(requestState),
        requestState.res.localTemplates.getAll(),
    );

    return {
        templates,
        palette,
        card: {
            ...card,
            variable_triggers: requestState.variableTriggers.getAll(),
            variables: [...requestState.variables],
        } as IDivData,
    };
};
