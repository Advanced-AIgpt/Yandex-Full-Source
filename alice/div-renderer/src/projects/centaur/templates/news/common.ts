import {
    ContainerBlock,
    DivBackground,
    FixedSize, IDivAction,
    ImageBackground,
    ImageBlock,
    MatchParentSize,
    SolidBackground,
    TextBlock,
    WrapContentSize,
} from 'divcard2';
import { compact } from 'lodash';
import { utcToZonedTime } from 'date-fns-tz';
import { differenceInCalendarDays, isSameDay } from 'date-fns';
import { Long } from 'protobufjs';
import { NAlice } from '../../../../protos';
import { formatDate } from '../../helpers/helpers';
import {
    colorDarkGrey1Opacity70,
    colorReadButton,
    colorWhite,
    colorWhiteOpacity60,
    offsetFromEdgeOfScreen,
} from '../../style/constants';
import { text40r, title32m, title60m } from '../../style/Text/Text';
import { getImageByTopic } from '../../helpers/newsHelper';
import { Avatar } from '../../../../common/helpers/avatar';

export const patchToOrig = (avatarsUrl: string) => avatarsUrl.replace(/\/[^/]+$/, '/orig');

export type NewsCardProps = {
    item: NAlice.NData.ITNewsItem;
    topic: string;
    bottomGap: boolean;
    tz: string;
};

export function newsCommonCard(
    { item: { Text, Image, Agency, Logo, PubDate, ExtendedNews }, topic, bottomGap, tz }: NewsCardProps,
    openExtendedNewsDivAction: IDivAction) {
    return new ContainerBlock({
        width: new MatchParentSize(),
        height: new MatchParentSize(),
        background: generateBackground(topic, Image),
        content_alignment_horizontal: 'left',
        content_alignment_vertical: 'bottom',
        paddings: {
            left: offsetFromEdgeOfScreen,
            bottom: bottomGap ? 104 : offsetFromEdgeOfScreen,
            right: offsetFromEdgeOfScreen,
        },
        items: [
            new ContainerBlock({
                width: new WrapContentSize(),
                height: new MatchParentSize({ weight: 1 }),
                content_alignment_vertical: 'bottom',
                orientation: 'horizontal',
                margins: {
                    bottom: 20,
                },
                items: newsSource(Agency, Logo, PubDate, tz),
            }),
            new ContainerBlock({
                width: new MatchParentSize(),
                height: new WrapContentSize(),
                content_alignment_vertical: 'bottom',
                orientation: 'horizontal',
                items: compact([
                    new TextBlock({
                        ...title60m,
                        width: new MatchParentSize({ weight: 1 }),
                        height: new WrapContentSize(),
                        max_lines: 5,
                        truncate: 'end',
                        text: Text ?? ' ',
                    }),
                    ExtendedNews?.length && new TextBlock({
                        ...text40r,
                        width: new WrapContentSize(),
                        height: new WrapContentSize(),
                        background: [new SolidBackground({
                            color: colorReadButton,
                        })],
                        border: {
                            corner_radius: 24,
                        },
                        paddings: {
                            top: 22,
                            right: 33,
                            bottom: 22,
                            left: 33,
                        },
                        margins: {
                            left: 48,
                        },
                        alignment_horizontal: 'right',
                        text: 'Читать',
                        action: openExtendedNewsDivAction,
                    }),
                ]),
            }),
        ],
    });
}

export const newsSource = (
    Agency: (string | null | undefined),
    Logo: (string | null | undefined),
    PubDate: (number | Long | null | undefined),
    tz: string,
) =>
    compact([
        Logo && new ContainerBlock({
            background: [
                new SolidBackground({
                    color: colorWhite,
                }),
            ],
            content_alignment_horizontal: 'center',
            content_alignment_vertical: 'center',
            border: {
                corner_radius: 12,
            },
            width: new FixedSize({ value: 40 }),
            height: new FixedSize({ value: 40 }),
            margins: {
                right: 20,
            },
            items: [
                new ImageBlock({
                    width: new FixedSize({ value: 28 }),
                    height: new FixedSize({ value: 28 }),
                    image_url: Logo ?? '',
                }),
            ],
        }),
        new TextBlock({
            ...title32m,
            text_color: colorWhiteOpacity60,
            width: new WrapContentSize(),
            height: new WrapContentSize(),
            text_alignment_vertical: 'center',
            text: (Agency ? Agency + ' • ' : '') + (Number(PubDate) ? formDate(Number(PubDate) * 1000 || 0, tz) : ''),
        }),
    ]);

export const generateBackground = (
    Topic: string | null | undefined,
    NewsImage: NAlice.NData.TNewsItem.ITImage | null | undefined,
    color = colorDarkGrey1Opacity70,
) => {
    const result: DivBackground[] = [];
    if (NewsImage && NewsImage.Src) {
        const avatar = Avatar.fromUrl(NewsImage.Src);

        result.push(
            new ImageBackground({
                image_url: avatar ? avatar.setTypeName('orig').toString() : NewsImage.Src,
                scale: 'fill',
                preload_required: 1,
            }),
        );
    } else {
        result.push(
            new ImageBackground({
                image_url: getImageByTopic(Topic),
                preload_required: 1,
            }),
        );
    }

    result.push(new SolidBackground({ color }));
    return result;
};

export const formDate = (date: number, tz: string) => {
    const d = utcToZonedTime(date, tz);
    const now = utcToZonedTime(Date.now(), tz);
    if (isSameDay(d, now)) {
        return formatDate(d, 'HH:mm');
    }
    if (differenceInCalendarDays(now, d) === 1) {
        return `вчера в ${formatDate(d, 'HH:mm')}`;
    }
    return formatDate(d, 'd MMMM в HH:mm');
};
