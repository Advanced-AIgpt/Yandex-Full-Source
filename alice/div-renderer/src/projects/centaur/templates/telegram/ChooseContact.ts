import {
    ContainerBlock,
    FixedSize,
    MatchParentSize,
    TextBlock,
    WrapContentSize,
    SolidBackground,
    Div,
} from 'divcard2';
import { compact } from 'lodash';
import { NAlice } from '../../../../protos';
import {
    colorWhiteOpacity50,
    colorWhiteOpacity90,
    offsetFromEdgeOfScreen,
    colorWhiteOpacity10,
    offsetTopWithSearchItem,
    simpleBackground,
} from '../../style/constants';
import { title32m, text40m } from '../../style/Text/Text';
import EmptyDiv from '../../components/EmptyDiv';
import { CloseButtonWrapper } from '../../components/CloseButtonWrapper/CloseButtonWrapper';
import { TopLevelCard } from '../../helpers/helpers';
import { directivesAction } from '../../../../common/actions';
import { telegramVideoCallTo } from '../../actions/server/telegramServerActions';
import { IRequestState } from '../../../../common/types/common';
import { MMRequest } from '../../../../common/helpers/MMRequest';

export const DeviceName = 'SmartSpeaker';

const header = new TextBlock({
    ...text40m,
    text_color: colorWhiteOpacity50,
    text: 'Уточните, кому будете звонить',
    height: new FixedSize({ value: 100 }),
    width: new WrapContentSize(),
    paddings: {
        top: 10,
        bottom: 10,
    },
});

export default function ChooseContact(
    chooseContactData: NAlice.NData.ITVideoCallContactChoosingData,
    _: MMRequest,
    requestState: IRequestState,
) {
    return TopLevelCard({
        log_id: 'choose_contact_card',
        states: [
            {
                state_id: 0,
                div: CloseButtonWrapper({
                    div: new ContainerBlock({
                        background: simpleBackground,
                        height: new MatchParentSize(),
                        width: new MatchParentSize(),
                        paddings: {
                            top: offsetTopWithSearchItem,
                            bottom: offsetFromEdgeOfScreen,
                        },
                        orientation: 'vertical',
                        content_alignment_vertical: 'top',
                        content_alignment_horizontal: 'left',
                        ...simpleBlock(chooseContactData.ContactData),
                    } as ConstructorParameters<typeof ContainerBlock>[0]),
                }),
            },
        ],
    }, requestState);
}

const simpleBlock = (
    contactsData: NAlice.NData.ITProviderContactData[] | null | undefined,
) => {
    return {
        items: compact([
            new EmptyDiv({
                width: new MatchParentSize(),
                height: new FixedSize({ value: 30 }),
            }),
            new ContainerBlock({
                paddings: {
                    left: offsetFromEdgeOfScreen,
                    right: offsetFromEdgeOfScreen,
                },
                margins: {
                    bottom: 20,
                },
                items: compact([
                    header,
                    contactItems(contactsData),
                ]),
            }),
        ]),
    };
};

const contactItems = (
    contactsData: NAlice.NData.ITProviderContactData[] | null | undefined,
) => {
    const contactItems: Div[] = [];
    if (contactsData !== undefined && contactsData !== null) {
        for (let i = 0; i < contactsData.length; i++) {
            const contactDiv = contactItem(contactsData[i].TelegramContactData);
            if (contactDiv !== undefined && contactDiv !== null) {
                contactItems.push(contactDiv);
                if (i + 1 < contactsData.length) {
                    contactItems.push(new EmptyDiv({
                        width: new FixedSize({ value: 100 }),
                    }));
                }
            }
        }
    }

    return new ContainerBlock({
        orientation: 'horizontal',
        content_alignment_vertical: 'top',
        content_alignment_horizontal: 'left',
        width: new MatchParentSize(),
        items: contactItems,
    });
};

const contactItem = (
    telegramContactData: NAlice.NData.TProviderContactData.ITTelegramContactData | null | undefined,
) => {
    return telegramContactData && new ContainerBlock({
        orientation: 'vertical',
        width: new FixedSize({ value: 300 }),
        height: new FixedSize({ value: 500 }),
        content_alignment_vertical: 'center',
        content_alignment_horizontal: 'center',
        border: {
            corner_radius: 30,
        },
        background: [
            new SolidBackground({ color: colorWhiteOpacity10 }),
        ],
        items: [
            new TextBlock({
                ...title32m,
                width: new WrapContentSize(),
                height: new WrapContentSize(),
                text_alignment_horizontal: 'center',
                text_alignment_vertical: 'center',
                margins: {
                    bottom: 50,
                },
                text_color: colorWhiteOpacity90,
                text: `${telegramContactData.DisplayName}`,
            }),
            telegramContactData && audioCallButton(telegramContactData),
            telegramContactData && videoCallButton(telegramContactData),
        ],
    });
};

const audioCallButton = (
    telegramContactData: NAlice.NData.TProviderContactData.ITTelegramContactData,
) => {
    return new ContainerBlock({
        orientation: 'vertical',
        width: new WrapContentSize(),
        content_alignment_vertical: 'center',
        content_alignment_horizontal: 'center',
        items: [
            new TextBlock({
                ...title32m,
                width: new FixedSize({ value: 250 }),
                height: new FixedSize({ value: 100 }),
                text_alignment_horizontal: 'center',
                text_alignment_vertical: 'center',
                text_color: colorWhiteOpacity90,
                border: {
                    corner_radius: 30,
                },
                margins: {
                    bottom: 20,
                },
                background: [
                    new SolidBackground({ color: colorWhiteOpacity10 }),
                ],
                text: 'Аудиозвонок',
            }),
        ],
        actions: compact([
            telegramContactData.UserId && {
                log_id: 'main_screen_video_call_to_action_without_video',
                url: directivesAction(telegramVideoCallTo(
                    telegramContactData.UserId,
                    telegramContactData.DisplayName,
                    false,
                )),
            },
        ]),
    });
};

const videoCallButton = (
    telegramContactData: NAlice.NData.TProviderContactData.ITTelegramContactData,
) => {
    return new ContainerBlock({
        orientation: 'vertical',
        width: new WrapContentSize(),
        content_alignment_vertical: 'center',
        content_alignment_horizontal: 'center',
        items: [
            new TextBlock({
                ...title32m,
                width: new FixedSize({ value: 250 }),
                height: new FixedSize({ value: 100 }),
                text_alignment_horizontal: 'center',
                text_alignment_vertical: 'center',
                text_color: colorWhiteOpacity90,
                border: {
                    corner_radius: 30,
                },
                margins: {
                    bottom: 20,
                },
                background: [
                    new SolidBackground({ color: colorWhiteOpacity10 }),
                ],
                text: 'Видеозвонок',
            }),
        ],
        actions: compact([
            telegramContactData.UserId && {
                log_id: 'main_screen_video_call_to_action_with_video',
                url: directivesAction(telegramVideoCallTo(
                    telegramContactData.UserId,
                    telegramContactData.DisplayName,
                    true,
                )),
            },
        ]),
    });
};
