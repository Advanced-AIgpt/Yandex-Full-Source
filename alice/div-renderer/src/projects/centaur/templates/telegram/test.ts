import { AnonymizeDataForSnapshot } from '../../../../common/helpers/dev';
import ChooseContact from './ChooseContact';
import { MMRequest } from '../../../../common/helpers/MMRequest';
import { createRequestState } from '../../../../registries/common';

jest.mock('../../../../common/logger');

describe('Telegram ChooseContact Scene', () => {
    it('matches snapshot', () => {
        const mmRequest = new MMRequest({}, {}, {});

        expect(AnonymizeDataForSnapshot(ChooseContact({
            ContactData: [{
                TelegramContactData: {
                    DisplayName: 'Name',
                    UserId: 'UserId',
                },
            }],
        }, mmRequest, createRequestState(mmRequest)))).toMatchSnapshot();
    });
});
