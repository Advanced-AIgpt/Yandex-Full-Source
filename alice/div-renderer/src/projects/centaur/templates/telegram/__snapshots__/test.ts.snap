// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Telegram ChooseContact Scene matches snapshot 1`] = `
Object {
  "card": Object {
    "log_id": "choose_contact_card",
    "states": Array [
      Object {
        "div": Object {
          "height": Object {
            "type": "match_parent",
            "weight": 1,
          },
          "items": Array [
            Object {
              "background": Array [
                Object {
                  "color": "#121316",
                  "type": "solid",
                },
              ],
              "content_alignment_horizontal": "left",
              "content_alignment_vertical": "top",
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [
                Object {
                  "delimiter_style": Object {
                    "color": "#0fff",
                  },
                  "height": Object {
                    "type": "fixed",
                    "value": 30,
                  },
                  "type": "separator",
                  "width": Object {
                    "type": "match_parent",
                  },
                },
                Object {
                  "items": Array [
                    Object {
                      "font_size": 40,
                      "font_weight": "medium",
                      "height": Object {
                        "type": "fixed",
                        "value": 100,
                      },
                      "line_height": 56,
                      "paddings": Object {
                        "bottom": 10,
                        "top": 10,
                      },
                      "text": "Уточните, кому будете звонить",
                      "text_color": "#80ffffff",
                      "type": "text",
                      "width": Object {
                        "type": "wrap_content",
                      },
                    },
                    Object {
                      "content_alignment_horizontal": "left",
                      "content_alignment_vertical": "top",
                      "items": Array [
                        Object {
                          "background": Array [
                            Object {
                              "color": "#1affffff",
                              "type": "solid",
                            },
                          ],
                          "border": Object {
                            "corner_radius": 30,
                          },
                          "content_alignment_horizontal": "center",
                          "content_alignment_vertical": "center",
                          "height": Object {
                            "type": "fixed",
                            "value": 500,
                          },
                          "items": Array [
                            Object {
                              "font_size": 32,
                              "font_weight": "medium",
                              "height": Object {
                                "type": "wrap_content",
                              },
                              "line_height": 40,
                              "margins": Object {
                                "bottom": 50,
                              },
                              "text": "Name",
                              "text_alignment_horizontal": "center",
                              "text_alignment_vertical": "center",
                              "text_color": "#e6ffffff",
                              "type": "text",
                              "width": Object {
                                "type": "wrap_content",
                              },
                            },
                            Object {
                              "actions": Array [
                                Object {
                                  "log_id": "main_screen_video_call_to_action_without_video",
                                  "url": "dialog://?directives=%5B%7B%22type%22%3A%22server_action%22%2C%22name%22%3A%22%40%40mm_semantic_frame%22%2C%22payload%22%3A%7B%22typed_semantic_frame%22%3A%7B%22video_call_to_semantic_frame%22%3A%7B%22fixed_contact%22%3A%7B%22contact_data%22%3A%7B%22telegram_contact_data%22%3A%7B%22display_name%22%3A%22Name%22%2C%22user_id%22%3A%22UserId%22%7D%7D%7D%2C%22video_enabled%22%3A%7B%22bool_value%22%3Afalse%7D%7D%7D%2C%22analytics%22%3A%7B%22product_scenario%22%3A%22VideoCall%22%2C%22origin%22%3A%22Scenario%22%2C%22purpose%22%3A%22alice_scenarios.video_call_to%22%2C%22origin_info%22%3A%22%22%7D%2C%22params%22%3A%7B%22disable_output_speech%22%3Atrue%2C%22disable_should_listen%22%3Atrue%7D%7D%7D%5D",
                                },
                              ],
                              "content_alignment_horizontal": "center",
                              "content_alignment_vertical": "center",
                              "items": Array [
                                Object {
                                  "background": Array [
                                    Object {
                                      "color": "#1affffff",
                                      "type": "solid",
                                    },
                                  ],
                                  "border": Object {
                                    "corner_radius": 30,
                                  },
                                  "font_size": 32,
                                  "font_weight": "medium",
                                  "height": Object {
                                    "type": "fixed",
                                    "value": 100,
                                  },
                                  "line_height": 40,
                                  "margins": Object {
                                    "bottom": 20,
                                  },
                                  "text": "Аудиозвонок",
                                  "text_alignment_horizontal": "center",
                                  "text_alignment_vertical": "center",
                                  "text_color": "#e6ffffff",
                                  "type": "text",
                                  "width": Object {
                                    "type": "fixed",
                                    "value": 250,
                                  },
                                },
                              ],
                              "orientation": "vertical",
                              "type": "container",
                              "width": Object {
                                "type": "wrap_content",
                              },
                            },
                            Object {
                              "actions": Array [
                                Object {
                                  "log_id": "main_screen_video_call_to_action_with_video",
                                  "url": "dialog://?directives=%5B%7B%22type%22%3A%22server_action%22%2C%22name%22%3A%22%40%40mm_semantic_frame%22%2C%22payload%22%3A%7B%22typed_semantic_frame%22%3A%7B%22video_call_to_semantic_frame%22%3A%7B%22fixed_contact%22%3A%7B%22contact_data%22%3A%7B%22telegram_contact_data%22%3A%7B%22display_name%22%3A%22Name%22%2C%22user_id%22%3A%22UserId%22%7D%7D%7D%2C%22video_enabled%22%3A%7B%22bool_value%22%3Atrue%7D%7D%7D%2C%22analytics%22%3A%7B%22product_scenario%22%3A%22VideoCall%22%2C%22origin%22%3A%22Scenario%22%2C%22purpose%22%3A%22alice_scenarios.video_call_to%22%2C%22origin_info%22%3A%22%22%7D%2C%22params%22%3A%7B%22disable_output_speech%22%3Atrue%2C%22disable_should_listen%22%3Atrue%7D%7D%7D%5D",
                                },
                              ],
                              "content_alignment_horizontal": "center",
                              "content_alignment_vertical": "center",
                              "items": Array [
                                Object {
                                  "background": Array [
                                    Object {
                                      "color": "#1affffff",
                                      "type": "solid",
                                    },
                                  ],
                                  "border": Object {
                                    "corner_radius": 30,
                                  },
                                  "font_size": 32,
                                  "font_weight": "medium",
                                  "height": Object {
                                    "type": "fixed",
                                    "value": 100,
                                  },
                                  "line_height": 40,
                                  "margins": Object {
                                    "bottom": 20,
                                  },
                                  "text": "Видеозвонок",
                                  "text_alignment_horizontal": "center",
                                  "text_alignment_vertical": "center",
                                  "text_color": "#e6ffffff",
                                  "type": "text",
                                  "width": Object {
                                    "type": "fixed",
                                    "value": 250,
                                  },
                                },
                              ],
                              "orientation": "vertical",
                              "type": "container",
                              "width": Object {
                                "type": "wrap_content",
                              },
                            },
                          ],
                          "orientation": "vertical",
                          "type": "container",
                          "width": Object {
                            "type": "fixed",
                            "value": 300,
                          },
                        },
                      ],
                      "orientation": "horizontal",
                      "type": "container",
                      "width": Object {
                        "type": "match_parent",
                      },
                    },
                  ],
                  "margins": Object {
                    "bottom": 20,
                  },
                  "paddings": Object {
                    "left": 48,
                    "right": 48,
                  },
                  "type": "container",
                },
              ],
              "orientation": "vertical",
              "paddings": Object {
                "bottom": 48,
                "top": 142,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
            Object {
              "alignment_horizontal": "right",
              "alignment_vertical": "top",
              "items": Array [
                Object {
                  "actions": Array [
                    Object {
                      "log_id": "close_fullscreen",
                      "url": "centaur://local_command?local_commands=[%7B%22command%22%3A%22close_layer%22%2C%22layer%22%3A%22DIALOG%22%2C%22stop_conversation%22%3Atrue%7D]",
                    },
                  ],
                  "alignment_horizontal": "right",
                  "background": Array [
                    Object {
                      "color": "#1affffff",
                      "type": "solid",
                    },
                  ],
                  "border": Object {
                    "corner_radius": 20,
                  },
                  "height": Object {
                    "type": "fixed",
                    "value": 88,
                  },
                  "image_url": "https://yastatic.net/s3/dialogs/smart_displays/icons/close_wt_border.png",
                  "margins": Object {
                    "right": 36,
                    "top": 36,
                  },
                  "paddings": Object {
                    "bottom": 22,
                    "left": 22,
                    "right": 22,
                    "top": 22,
                  },
                  "preload_required": 1,
                  "scale": "fill",
                  "type": "image",
                  "width": Object {
                    "type": "fixed",
                    "value": 88,
                  },
                },
              ],
              "type": "container",
            },
          ],
          "orientation": "overlap",
          "type": "container",
          "width": Object {
            "type": "match_parent",
            "weight": 1,
          },
        },
        "state_id": 0,
      },
    ],
    "variable_triggers": Array [],
    "variables": Array [],
  },
  "templates": Object {},
}
`;
